handler/src/main/java/io/netty/handler/stream/ChunkedNioFile.java

```diff
@@ -101,9 +102,8 @@ public ChunkedNioFile(FileChannel in, long offset, long length, int chunkSize)
                     "chunkSize: " + chunkSize +
                     " (expected: a positive integer)");
         }
-
-        if (offset != 0) {
-            in.position(offset);
+        if (!in.isOpen()) {
+            throw new ClosedChannelException();
```

Comment:
@franz1981 I wonder if we should revert this as it basically change the behaviour in a bug-fix release.

Reply:
It changes the behaviour indeed and I don't think that ChunkedNioFile has a  real"bug" ATM: probably is just not an "expected" behaviour.
I see that ChunkedFile is the same, while `DefaultFileRegion` not: that's why I'm providing this fix..
Beside an additional handler in the pipeline, a user should use `DefaultFileRegion` and `ChunkedNioFIle` in the same exact way, delivering the same functionality, just with a difference on perf side. This PR allows to do that, while the original code not...

Reply:
I see... can you add a unit test for it as well and add some details to the commit message ? As this is not really related to the change descripted in the issue I wonder if it may even be better to make it a separate PR. I think I would prefer this.

Reply:
Wait a sec, but `in.position(offset);` will throw `ClosedChannelException();` as well, so my code ha replicated the internal check...or do you mean another thing?

Reply:
oh cool... didn't realise that :) So good as it is then. Just please add a unit test :)

Reply:
Added another test to stress the `ClosedChannelException` case